src/java.base/share/classes/sun/security/ssl/HandshakeContext.java

```diff
@@ -164,7 +164,7 @@ protected HandshakeContext(SSLContextImpl sslContext,
         this.conContext = conContext;
         this.sslConfig = (SSLConfiguration)conContext.sslConfig.clone();
 
-        this.algorithmConstraints = new SSLAlgorithmConstraints(
+        this.algorithmConstraints = SSLAlgorithmConstraints.wrap(
```

Comment:
Maybe, the change could be placed in the SSLAlgorithmConstraints constructors implementation so that it is easier to avoid this mistake.